src/measure_theory/card_measurable_space.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2022 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import measure_theory.measurable_space_def
+import set_theory.continuum
+import set_theory.cofinality
+
+/-!
+# Cardinal of sigma-algebras
+
+If a sigma-algebra is generated by a set of sets `s` whose cardinality is at most the continuum,
+then the sigma-algebra satisfies the same cardinality bound. This is stated in
+`measurable_space.cardinal_generate_measurable_le` and
+`measurable_space.cardinal_measurable_set_le`.
+
+For the proof, we rely on an explicit inductive construction of the sigma-algebra generated by
+`s` (instead of the inductive predicate `generate_measurable`). This transfinite inductive
+construction is parameterized by an ordinal `< ℵ₁`, and the cardinality bound is preserved along
+each step of the construction.
+-/
+
+universe u
+variables {α : Type u}
+
+open_locale cardinal
+open cardinal set
+
+local notation a `<₁` b := (aleph 1 : cardinal.{u}).ord.out.r a b
+local notation `ℵ₁`:= (aleph 1 : cardinal.{u}).ord.out.α
+
+namespace measurable_space
+
+/-- Transfinite induction construction of the sigma-algebra generated by a set of sets `s`. At each
+step, we add all elements of `s`, the empty set, the complements of already constructed sets, and
+countable unions of already constructed sets. We index this construction by an ordinal `< ℵ₁`, as
+this will be enough to generate all sets in the sigma-algebra. -/
+def generate_measurable_rec (s : set (set α)) : ℵ₁ → set (set α)
+| i := s ∪ {∅} ∪ (λ (t : set α), tᶜ) '' (⋃ j : {j // j <₁ i}, generate_measurable_rec j.1)
+    ∪ (set.range (λ (f : ℕ → (⋃ j : {j // j <₁ i}, generate_measurable_rec j.1)),
+        (⋃ n, (f n).1)))
+using_well_founded {dec_tac := `[exact j.2]}
+
+/-- At each step of the inductive construction, the cardinality bound `≤ 𝔠` is preserved. -/
+lemma cardinal_generate_measurable_rec_le_aux {s : set (set α)} (hs : #s ≤ 𝔠) (i : ℵ₁) :
+  #(generate_measurable_rec s i) ≤ 𝔠 :=
+begin
+  apply well_founded.induction (aleph 1 : cardinal.{u}).ord.out.wo.wf i,
+  assume i IH,
+  have J : #(⋃ (j : {j // j <₁ i}), generate_measurable_rec s j.1) ≤ 𝔠,
+  { apply (mk_Union_le _).trans,
+    have A : # {j // j <₁ i} ≤ aleph 1 := (mk_subtype_le _).trans (le_of_eq (aleph 1).mk_ord_out),
+    have B : (cardinal.sup.{u u} (λ (j : {j // j <₁ i}), #(generate_measurable_rec s j.1)))
+                ≤ 𝔠,
```

Comment:
Did you accidentally hit enter?

Reply:
No, otherwise the line is too long by one character...

Reply:
It's only 96 characters long.

Reply:
Ah, it used to be too long with the old name of `generate_measurable_rec`, but since then I changed the name and now it fits in a line. Thanks!